dvc/api.py

```diff
@@ -1,16 +1,42 @@
+import importlib
 import os
+import sys
+import copy
 from contextlib import contextmanager
 
 try:
     from contextlib import _GeneratorContextManager as GCM
 except ImportError:
     from contextlib import GeneratorContextManager as GCM
 
-from dvc.utils.compat import urlparse
+import ruamel.yaml
+from voluptuous import Schema, Optional, Invalid, ALLOW_EXTRA
+
+from dvc.utils.compat import urlparse, builtin_str
 from dvc.repo import Repo
+from dvc.exceptions import SummonError, FileMissingError
 from dvc.external_repo import external_repo
 
 
+summon_schema = Schema(
+    {
+        "objects": [
+            {
+                "name": str,
+                "summon": {
+                    "type": "python",
+                    "call": str,
+                    Optional("args"): dict,
+                    Optional("deps"): [str],
+                },
+            }
+        ]
+    },
+    required=True,
+    extra=ALLOW_EXTRA,
```

Comment:
We should not allow extra everywhere. May not allow it anywhere for now for better future-compatibility. Say only allow `objects` at the top, and only allow `name`, `meta` and `summon` in the object, all the user data should go under `meta`, like description, paper, author, etc.